### ISSUE TITLE
Occasionally after unlocking the screen via swaylock or something else, the compositor is unable to give keyboard focus to a surface

### ISSUE THREAD

**Issue by mattkae** [canonical/mir#3269]
## Repro
This only happens occasionally

1. Have a terminal opened
2. Lock the screen using swaylock
3. Unlock the screen using swaylock
4. Try to type something in the terminal, but note that the text isn't coming through
5. If you can, switch workspaces and note that text focus returns
